crates/cheatnet/src/cheatcodes/deploy.rs

```diff
@@ -25,15 +25,39 @@ impl CheatnetState {
         &mut self,
         class_hash: &ClassHash,
         calldata: &[Felt252],
+        predefined_contract_address: Option<ContractAddress>,
```

Comment:
I would create another method on `CheatnetState` and use composition, instead of adding a parameter here:
Public methods:
`deploy_at(address)` -> deploys or panics if address is taken
`deploy()` -> generates address for you, uses `deploy_at` afterwards